### PR TITLE
Feature/get data

### LINKED ISSUE #9: Iteration 2

### LINKED ISSUE #20: User interactivity with Homepage
User Stories (Dashboard)
Use the scripts.js file to add information to the DOM. This JS file should call methods from your classes to retrieve data. There should not be any DOM manipulation within your class files.

As a user, I should be able to view a list of all recipes.
As a user, I should be able to click on a recipe to view more information including directions, ingredients needed, and total cost.
As a user, I should be able to filter recipes by a tag. (Extension option: by multiple tags)
As a user, I should be able to search recipes by their name. (Extension option: by name or ingredients)

### LINKED ISSUE #32: Random user on load

### LINKED ISSUE #33: Can see and change user saved recipes
User Data
Create classes and methods that can:

Allow a user to add/remove a recipe to their recipesToCook list (add to my recipesToCook)
Filter my recipesToCook by a tag. (Extension option: filter by multiple tags)
Filter my recipesToCook by its name. (Extension option: filter by name or ingredients)
User Stories
On load, a user should be chosen at random.

As a user, I should be able to add/remove a recipe to a list of recipes to cook
As a user, I should be able to filter my toCook recipes by a tag. (Extension option: filter by multiple tags)
As a user, I should be able to search my toCook recipes by its name. (Extension option: search by name or ingredients)

### LINKED ISSUE #34: Allow user to view saved recipes

### LINKED ISSUE #35: Allow user to add/remove saved recipes

### LINKED ISSUE #36: Allow user to filter saved recipes

### LINKED ISSUE #37: Allow user to search saved recipes

### LINKED ISSUE #38: recipesToCook[]

### LINKED ISSUE #41: addRemoveRecipe()

### LINKED ISSUE #47: Data should all be imported

### LINKED ISSUE #48: Delete hard-coded data

### LINKED ISSUE #49: Get all users

### LINKED ISSUE #50: Get all ingredients

### LINKED ISSUE #51: Get all recipes

### LINKED ISSUE #54: Allow users to to search all/filtered/saved recipe matches for each letter keystroke

### LINKED ISSUE #65: Get rid of "User Info" placeholder
On page load, there's a brief instance where the header says "user info" - just want to get rid of that.

### PR DESCRIPTION
**Pull request by jfogiato**
Files Changed: apiCalls.js, Recipe.js., ingredients.js (removed), recipes.js (removed), users.js (removed), test-data.js (added), scripts.js, Recipe-test.js, RecipeRepo-test.js, User-test.js

Change Summary: Added full Fetch API functionality. Added a separate `test-data.js` file for all tests. Updated tests to utilize the test file. Cleaned up imports in scripts.js. Refactored various functions to utilize more parameters and global view variables. Update save/remove button to change dynamically after click. Update search to filter as you type. 

Other Notes: HOME STRETCH!

Closes [#61](https://github.com/jfogiato/whats-cookin/issues/61), Closes [#58](https://github.com/jfogiato/whats-cookin/issues/58), Closes [#57](https://github.com/jfogiato/whats-cookin/issues/57), Closes [#45](https://github.com/jfogiato/whats-cookin/issues/45), Closes [#42](https://github.com/jfogiato/whats-cookin/issues/42), Closes [#41](https://github.com/jfogiato/whats-cookin/issues/41), Closes [#38](https://github.com/jfogiato/whats-cookin/issues/38), Closes [#37](https://github.com/jfogiato/whats-cookin/issues/37), Closes [#36](https://github.com/jfogiato/whats-cookin/issues/36), Closes [#35](https://github.com/jfogiato/whats-cookin/issues/35), Closes [#34](https://github.com/jfogiato/whats-cookin/issues/34), Closes [#33](https://github.com/jfogiato/whats-cookin/issues/33), Closes [#32](https://github.com/jfogiato/whats-cookin/issues/32), Closes [#20](https://github.com/jfogiato/whats-cookin/issues/20), Closes [#9](https://github.com/jfogiato/whats-cookin/issues/9), Closes #54 , Closes #49 , Closes #50 , Closes #51 , Closes #48 , Closes #47 , Closes #65

### CODE REVIEW

File: src/classes/Recipe.js [hunk 1]
```diff
@@ -47,7 +47,7 @@ class Recipe {
   }
 
   getInstructions() {
-    const instructions = this.instructions.map(step => `${step.instruction}`)
+    const instructions = this.instructions.map(step => `${step.instruction}`);
```
Comment (qrispi) [inline src/classes/Recipe.js:50]: Those darn semi-colons!

**Comment by LordSchwifty** [comment]
A lot of good work here. Solid refactoring in scripts and the different classes!

File: src/scripts.js [hunk 2]
```diff
@@ -17,42 +13,34 @@ const searchBar = document.getElementById('searchBar');
...
-  }
-});
-
 logo.addEventListener('click', goHome);
-
-
-
-
-
+searchBar.addEventListener('keyup', searchRecipes);
```
Comment (qrispi) [inline src/scripts.js:31]: Love how much cleaner this whole section is!

File: src/scripts.js [hunk 3]
```diff
@@ -17,42 +13,34 @@ const searchBar = document.getElementById('searchBar');
...
-
-const recipeRepo = new RecipeRepository(recipeData);
+let users;
+let ingredients;
+let recipes;
+let recipeRepo;
 let modalRecipe;
 let currentUser;
 let savedView = false;
+let currentView;
```
Comment (jfogiato) [inline src/scripts.js:23]: S/o @qrispi for this - brilliant solution!

File: src/scripts.js (hunk 2 above)
Comment (jfogiato) [inline src/scripts.js:31]: S/o @jsahim on this one, too!!

Comment (qrispi) [inline src/scripts.js:31]: @jsahim This worked so well and was so much easier to implement! Great work!

File: src/scripts.js [hunk 4]
```diff
@@ -17,42 +13,34 @@ const searchBar = document.getElementById('searchBar');
...
 logo.addEventListener('click', goHome);
-
-
-
-
-
+searchBar.addEventListener('keyup', searchRecipes);
 
 //functions
+apiCalls().then(data => {
```
Comment (jfogiato) [inline src/scripts.js:34]: Great work to the team for figuring this one out

File: src/scripts.js (hunk 3 above)
Comment (LordSchwifty) [reply src/scripts.js:23]: This was very creative! Looks much cleaner.

File: test/Recipe-test.js [hunk 5]
```diff
@@ -1,203 +1,16 @@
...
-      "tags": [
-        "antipasti",
-        "starter",
-        "snack",
-        "appetizer",
-        "antipasto",
-        "hor d'oeuvre"
-      ]
-    }
+    recipeData = testData.recipes;
```
Comment (qrispi) [inline test/Recipe-test.js:9]: Great work cleaning up the test data files!

**Review by qrispi — APPROVED** [review]
Love all these changes, time to have some fun making everything look cool!

File: src/scripts.js (hunk 4 above)
Comment (LordSchwifty) [reply src/scripts.js:34]: Great work on apiCalls(). Functioning perfectly!

**Pull request merged**